### PR TITLE
restore compatibility with 2.0.0's fix for wraparound bug

### LINKED ISSUE #80: Possible regression of "BSD wraparound" bug?
[passlib implements a specific check](https://bitbucket.org/ecollins/passlib/src/934db60eea9cd9fe16fed88dcac114435256af47/passlib/handlers/bcrypt.py#bcrypt.py-318) against the ["BSD wraparound" bug](http://www.openwall.com/lists/oss-security/2012/01/02/4) that now appears to be failing with bcrypt 3.0

Although the included source from libcrypt appears to have been updated since the bug was fixed ([OpenBSD 5.5, according to passlib](http://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt.html#bsd-wraparound-bug)) I'm not familiar enough with the code to be able to tell if this is a genuine regression or not

### PR DESCRIPTION
**Pull request by reaperhulk**
fixes #80

### CODE REVIEW

**Comment by reaperhulk** [comment]
This fix is a bit weird to me since arguably OpenBSD's "If you're on the old $2a$ hash you may want this behavior because you're checking existing hashes that were made with that behavior" is more correct, but I guess preserving compatibility with ourselves is more important.

**Comment by dstufft** [comment]
@reaperhulk Yea-- I think it's especially important to break those particular hashes because they're trivial to brute force (since they hash the same as ``b"0" * 72``) which is one of the first things that something like JackTheRipper will try. However, given that we had the old behavior previously, if we didn't restore it we'd also be breaking anyone who happened to use this kind of password (although the edge case this hits it's quite rare).

Ultimately though, I think the edge case is sufficiently narrow so as not to matter *that* much, but the fact it generates a warning in passlib motivates me to ensure this is fixed :)

File: tests/test_bcrypt.py [hunk 1]
```diff
@@ -375,3 +375,9 @@ def test_invalid_params(password, salt, desired_key_bytes, rounds, error):
 def test_bcrypt_assert():
     with pytest.raises(SystemError):
         bcrypt._bcrypt_assert(False)
+
+
+def test_2a_wraparound_bug():
+    assert bcrypt.hashpw(
+        (b"0123456789" * 26)[:255], b"$2a$04$R1lJ2gkNaoPGdafE.H.16."
+    ) != b"$2a$04$R1lJ2gkNaoPGdafE.H.16.nVyh2niHsGJhayOHLMiXlI45o8/DU.6"
```
Comment (charmander) [inline tests/test_bcrypt.py]: May as well make a positive assertion here?

Comment (reaperhulk) [inline tests/test_bcrypt.py]: Done :)
